### PR TITLE
Update project 1 starter code

### PR DESCRIPTION
**Pull request by jkosh44**
TODO:
- [x] Update new class name
- [x] Add comments to new class
- [x] Add tests for new class

### CODE REVIEW

**Review by mbutrovich — APPROVED** [review]
LGTM

File: src/include/buffer/buffer_pool_manager_instance.h [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  BufferPoolManagerInstance(size_t pool_size, DiskManager *disk_manager, LogManager *log_manager = nullptr);
+
+  /**
+   * Destroys an existing BufferPoolManagerInstance.
+   */
+  ~BufferPoolManagerInstance() override;
+
+  /** Grading function. Do not modify! */
+  Page *FetchPage(page_id_t page_id, bufferpool_callback_fn callback = nullptr) override {
+    GradingCallback(callback, CallbackType::BEFORE, page_id);
```
Comment (jkosh44) [inline src/include/buffer/buffer_pool_manager_instance.h:48]: It's probably possible to push these functions into the parent class and make the *Impl functions virtual. That way it's not duplicated across the sub-classes. Thoughts?

File: src/include/buffer/buffer_pool_manager_pool.h [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "buffer/buffer_pool_manager.h"
+
+namespace bustub {
+
+class BufferPoolManagerPool : public BufferPoolManager {
```
Comment (jkosh44) [inline src/include/buffer/buffer_pool_manager_pool.h:19]: I think this class needs a better name

File: src/include/buffer/buffer_pool_manager_pool.h [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    }
+  }
+
+  /**
+   * @param page_id id of page
+   * @return pointer to the BufferPoolManager responsible for handling given page id
+   */
+  BufferPoolManager *GetBufferPoolManager(page_id_t page_id);
+
+  /**
```
Comment (jkosh44) [inline src/include/buffer/buffer_pool_manager_pool.h:106]: These comments need to be updated.

**Pull request merged**